internal/alloycli/cmd_run.go

```diff
@@ -192,6 +195,12 @@ func (fr *alloyRun) Run(configPath string) error {
 
 	level.Info(l).Log("boringcrypto enabled", boringcrypto.Enabled)
 
+	// Set the memory limit, this will honor GOMEMLIMIT if set
+	// If there is a cgroup will follow that
+	if fr.minStability.AtLeast(featuregate.StabilityPublicPreview) {
+		memlimit.SetGoMemLimitWithOpts(memlimit.WithLogger(slog.Default()))
```

Comment:
I think this should use `slog.New(l.Handler())` here instead of `slog.Default()`; the controller's logger implements the slog.Handler interface. I think this is important so that log entries are emitted in the proper format if the user requests json vs the default.

Reply:
I tried that. The format isnt set by this point so it errors, its possible we could default to one?

Reply:
I could always move it after the logger is resolved but that feels like an odd time to set memory limits.

Reply:
> I think it might be worth fixing that so we can use the right logger here.

Since we've fixed issues about this in the past I'd say not using the right logger would be a regression here. IMO I think we need to do something to use the right logger, either fixing the buffering issue with the slog.Handler implementation or moving this later in evaluation at least temporarily (I agree it's weird for setting the memory limit to not happen as early as possible)

Reply:
I am going to fork this out into its own PR, making the logger work with slog is bigger than this change.

Reply:
@mattdurham Is creating a PR to fix the logger still on your radar? Given this was merged first, there's a known regression in main now, and we're running out of time for 1.1 (RC gets cut this Thursday). 

If we can't fix this by Thursday we'll have to discuss this feature skipping 1.1 or putting out the release with a known regression.

Reply:
It is, wrapping up testing now, it ended up being slightly more complicated than expected.